File: 334.c
#include "shdstd.h"

#define MAX(x, y) (((x) > (y)) ? (x) : (y))
#define MIN(x, y) (((x) < (y)) ? (x) : (y))

#define MOD 1000000007

bool increasingTriplet(int *nums, int size)
{
	int most_min = INT_MAX;
	int min = INT_MAX;

	int i;
	for (i = 0; i < size; ++i) {
		int num = nums[i];
		most_min = MIN(most_min, num);
		if (most_min == num) {
			continue;
		}
		min = MIN(min, num);
		if (min == num) {
			continue;
		}
		return true;
	}

	return false;
}

File: 334.go
package main

import (
	"fmt"
	"math"
)

func increasingTriplet(nums []int) bool {
	most_min := math.MaxInt
	second_min := math.MaxInt

	for i := 0; i < len(nums); i++ {
		num := nums[i]
		most_min = min(most_min, num)
		if most_min == num {
			continue
		}

		second_min = min(second_min, num)
		if second_min == num {
			continue
		}

		return true
	}

	return false
}

func main() {
	nums := []int{1, 2, 3, 4, 5}
	res := increasingTriplet(nums)
	fmt.Println(res)
}
